From c54daa661ec492504c069cd39d9b592cd4657cd2 Mon Sep 17 00:00:00 2001
From: Srinivas Neeli <srinivas.neeli@xilinx.com>
Date: Wed, 28 Aug 2019 19:21:00 +0530
Subject: [PATCH] gpio: gpio-xilinx: Fix error condition for defer case

commit 73c42b232a896e9aaa6050e9f99d97f88e0dc762 from
https://github.com/Xilinx/linux-xlnx.git

Error condition always true, regardless of the error status indicated
by the call PTR_ERR(chip->clk). In Defer case, should not print the
error message. This patch adding a check for defer case.

Signed-off-by: Srinivas Neeli <srinivas.neeli@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Haiqing Bai <Haiqing.Bai@windriver.com>
---
 drivers/gpio/gpio-xilinx.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpio/gpio-xilinx.c b/drivers/gpio/gpio-xilinx.c
index 66aeef9b6101..decfdcf7cedc 100644
--- a/drivers/gpio/gpio-xilinx.c
+++ b/drivers/gpio/gpio-xilinx.c
@@ -649,9 +649,9 @@ static int xgpio_of_probe(struct platform_device *pdev)
 
 	chip->clk = devm_clk_get(&pdev->dev, "s_axi_aclk");
 	if (IS_ERR(chip->clk)) {
-		if ((PTR_ERR(chip->clk) != -ENOENT) ||
-				(PTR_ERR(chip->clk) != -EPROBE_DEFER)) {
-			dev_err(&pdev->dev, "Input clock not found\n");
+		if (PTR_ERR(chip->clk) != -ENOENT) {
+			if (PTR_ERR(chip->clk) != -EPROBE_DEFER)
+				dev_err(&pdev->dev, "Input clock not found\n");
 			return PTR_ERR(chip->clk);
 		}
 
-- 
2.17.1

